pkg/cmd/enable/addons/exec.go

```diff
@@ -0,0 +1,134 @@
+// Copyright Contributors to the Open Cluster Management project
+package addons
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	clusterclientset "open-cluster-management.io/api/client/cluster/clientset/versioned"
+	"open-cluster-management.io/clusteradm/pkg/cmd/enable/addons/scenario"
+	"open-cluster-management.io/clusteradm/pkg/helpers"
+	"open-cluster-management.io/clusteradm/pkg/helpers/apply"
+)
+
+const appMgrAddonName = "application-manager"
+
+//ClusterName: The cluster name used in the template
+type ClusterName struct {
+	ClusterName string
+}
+
+func (o *Options) complete(cmd *cobra.Command, args []string) (err error) {
+	klog.V(1).InfoS("addon options:", "dry-run", o.ClusteradmFlags.DryRun, "names", o.names, "clusters", o.clusters, "output-file", o.outputFile)
+
+	return nil
+}
+
+func (o *Options) validate() error {
+	if o.names == "" {
+		return fmt.Errorf("names is missing")
+	}
+
+	names := strings.Split(o.names, ",")
+	for _, n := range names {
+		if n != appMgrAddonName {
+			return fmt.Errorf("invalid add-on name %s", n)
+		}
+	}
+
+	if o.clusters == "" {
+		return fmt.Errorf("clusters is misisng")
+	}
+
+	return nil
+}
+
+func (o *Options) run() error {
+	alreadyProvidedAddons := make(map[string]bool)
+	addons := make([]string, 0)
+	names := strings.Split(o.names, ",")
+	for _, n := range names {
+		if _, ok := alreadyProvidedAddons[n]; !ok {
+			alreadyProvidedAddons[n] = true
+			addons = append(addons, strings.TrimSpace(n))
+		}
+	}
+	o.values.addons = addons
+
+	alreadyProvidedClusters := make(map[string]bool)
```

Comment:
this seems not used?

Reply:
This is used to filter out duplicated cluster names. For example in the command below, `cluster1` is repeated 3 times but after filtering only 1 value is being used.
```
$ clusteradm enable addons --names application-manager --clusters cluster1,cluster1,cluster1 -v 3
I0720 11:29:14.378189   15685 exec.go:30] "addon options:" dry-run=false names="application-manager" clusters="cluster1,cluster1,cluster1" output-file=""
I0720 11:29:14.378239   15685 exec.go:77] "values:" addons=[application-manager] clusters=[cluster1]
```

Reply:
I see, I think you can use set.String.